boot/zephyr/CMakeLists.txt

```diff
@@ -213,5 +213,19 @@ if(NOT CONFIG_BOOT_SIGNATURE_KEY_FILE STREQUAL "")
     DEPENDS ${KEY_FILE}
     )
   zephyr_library_sources(${GENERATED_PUBKEY})
+
+  set(GENERATED_PRIVKEY ${ZEPHYR_BINARY_DIR}/autogen-privkey.c)
+  add_custom_command(
+      OUTPUT ${GENERATED_PRIVKEY}
+    COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${MCUBOOT_DIR}/scripts/imgtool.py
+    getpriv
+    -k
+    ${KEY_FILE}
+    > ${GENERATED_PRIVKEY}
+    DEPENDS ${KEY_FILE}
+    )
+  zephyr_library_sources(${GENERATED_PRIVKEY})
```

Comment:
As I understand, we're changing the behavior from every build using the same hardcoded key, to generating a new one that varies for each build directory this is run in. If I were to clean and rebuild, I would get another key. Aren't we going to need a little better key management to coordinate these keys to whoever is making the build?

Reply:
The current patch hands out the responsibility of the key generation (when and how often) to the user. This could be documented. However, if we were to generate a new private key on each build, we should add an extra command here doing this job.

Reply:
Except that the user won't likely be aware of it. The first time they build, they will get a new key. If they clean their tree and rebuild, they will get a new key. @utzig any thoughts on this?